sdcm/sct_runner.py

```diff
@@ -600,8 +602,67 @@ def _create_instance(self,
         LOGGER.info("Got public IP: %s", instance.public_ip_address)
         self.instance = instance
 
+        # If the SCT runner instance is used as a Docker backend for tests
+        if os.getenv("SCT_CLUSTER_BACKEND") == "docker":
+            LOGGER.info("Configure unused disks to use as a persistent storage for Docker.")
+            LOGGER.info("Connecting to instance...")
+            remoter = self.get_remoter(host=instance.public_ip_address, connect_timeout=120)
+
+            LOGGER.info("Getting unused disks...")
+            disks = self._get_unused_disks(remoter)
+            LOGGER.info("The following unused disks are detected: %s", disks)
+            if disks:
+                mount_point = "/mnt/docker-persistent-storage"
+                self._setup_disks(remoter, disks, mount_point)
+                LOGGER.info("Disks are configured.")
+                self._set_docker_data_root(remoter, mount_point)
+                LOGGER.info("Persistent storage for Docker is configured.")
+            else:
+                LOGGER.info("No unused disks found.")
+
+            remoter.stop()
+
         return instance
 
+    @staticmethod
+    def _get_unused_disks(remoter: Any) -> list[str]:
+        result = remoter.run("lsblk -dnpo NAME,TYPE | grep disk | cut -d' ' -f1")
+        devices = result.stdout.strip().splitlines()
+
+        unused = []
+        for device in devices:
+            dev = device.replace("/dev/", '')
+            if len(glob.glob("/sys/class/block/{dev}/{dev}*".format(dev=dev))) == 0:
+                unused.append(device)
+        return unused
+
+    @staticmethod
+    def _setup_disks(remoter: Any, disks: list, mount_point: str) -> None:
```

Comment:
Disk setup doesn't look as a requirement for being able to run a test on docker backend.
We do not expect perf to be great in this case.

Moreover, commit topic and description doesn't tell anything about the disk setup.
About such a significant change as changing disks formatting...

Reply:
Without a dedicated physical disk Nemesis cases are always failing, when executed on Docker backend in Jenkins.
Only after configuring a free disk (which comes with e.g. an i4i instance type) to be used as a permanent storage for Docker - the Nemesis cases started to pass.

Reply:
If nemesis that depends on the disk perf fails all the time without local NVMes then there should be detection of it and proper skipping.

Reply:
Updated the commit message.
With regard to the configuring the dedicated disk - no Nemesis case is able to run on EBS, So the dedicated disk is necessary to be able to run docker backend on AWS.